### ISSUE TITLE
Hybrid assembly with contigs plus long reads

### ISSUE THREAD

**Issue by mesamuels** [fenderglass/ABruijn#635]
Apologies if this has been covered, if so please point me to the right place. I want to do a hybrid assembly with high coverage Illumina short insert reads and lower coverage ONT reads. My colleague says Flye will accept pre-assembled contigs made from just the Illumina reads (with Platanus), together with the ONT fastq reads, with the goal to try and merge contigs that are split by repetitive sequences. This doesn't appear to be a work flow generally used, and I haven't been able to figure out exactly how to format a Flye command line with these sorts of inputs, and a rough guess at the parameters (genome size is about 1 gigabase). Is it the case that Flye can take a contig file in fasta format and an ONT file in fastq format, and if so how do I basically format the command line for those input files?

If not, can I use the paired forward and reverse Illumina fastq files and the single ONT fastq file in one de novo assembly step?

Thanks in advance!
Mark Samuels
Associate Professor in Medicine
University of Montreal

BTW, we published the highly fragmented assembly from just the Illumina reads: https://www.mdpi.com/2223-7747/12/3/467

**Comment by margaretc-ho** [comment]
@fenderglass
+1 I am also interested in this method of hybrid assembly as well (we have pacbio, ONT and illumina). It is possible to use pre-assembled Illumina contigs together with long reads with Flye? Thanks

**Comment by mesamuels** [comment]
I saw your comment. I haven't heard back, but there is a tool that uses Flye as part of its pipeline. It's called Samba, here's the citation: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8849508/ 

I was able to install and run it. It took as input a fasta file of my Illumina short read scaffolds, plus a fastq file of ONT reads. It took less than 24 hours to run using 64 Gb of memory, however my ONT file was pretty small as it was just a pilot run, only 1Gb or so of total sequence. Undoubtedly it would take longer if your long read file is big, but no idea how the time scales. Also there are some other tools that are designed to do the same kind of merge. I have installed LongStitch (https://www.ncbi.nlm.nih.gov/pmc/?term=longstitch) but haven't tried it yet. Good luck!

**Comment by margaretc-ho** [comment]
Thanks @mesamuels I am also running hybrid assemblies with MASURCA, Wengan and HASLR with PB, ONT, and Illumina, as well as Canu with both PB and ONT reads, then polishing with Illumina later. Want to compare these different assembly methods together. I figured I would piggyback on your comment to ask if Flye has the capability to do similar analyses, but perhaps not.

p.s Longstitch corrects for misassembly and scaffolds but it doesn't actually gapfill successfully (probably a bug), so I had to use other tools to fill in the gaps that it left in an assembly I ran it on. SAMBA is a great tool.

**Comment by mesamuels** [comment]
Thanks Margaret, I see you know about Samba, if you look at the paper it mentions that some components of Flye are used, so maybe that is what my colleague here was originally thinking of.

I haven't heard of Wengan, I will look into it.

When you say LongStitch doesn't "gap fill", not sure what you mean. I imagine that my short read scaffolds mostly have repetitive sequences at their ends; the long reads would be to bridge them using unique regions on both sides of the particular repeat. Filling the gap of repetitive sequence itself is not that important for me. My problem is that many genes are fragmented on different scaffolds, so I have to build gene models by hand one-by-one, by Blasting the assembly and finding the various scaffolds that contain the various exons, and concatenating them manually. Possible, but very laborious and can't be automated for comparing large numbers of gene sequences with related species.

Mark

**Comment by LukeHebert** [comment]
Just wanted to comment to say I'm interested in the same feature. I have a set of large "genomic scaffolds" that make up the current genome assembly for the mammal I'm studying and would like to place these scaffolds relative to one another using a Flye-assembled "genome" (it can be a rough assembly) built from Oxford Nanopore Technology bulk genomic DNA reads.

In the meantime, I'll look into LongStitch!

**Comment by mesamuels** [comment]
Luke, thanks for the response. I have not heard anything from the Flye group, however I think Flye is incorporated into a pipeline called SAMBA. See Zimin & Salzberg PLoS Comput Biol 18(2): e1009860. I was able to install it on our Compute Canada national system and run it, and it did something, still analyzing the result. I did not have good long read data however, there was a problem with my (plant) DNA so the Oxford Nano reads were still pretty short, so I wouldn't expect them to help much. In terms of general statistics, there are a new new long scaffolds and the average scaffold size got a little bit bigger.

Someone said that LongStitch did not fill in gaps, it just links scaffolds together. I haven't tried it yet, although I was able to install it on our system.

Good luck!
Mark Samuels
________________________________
